client/gutenberg/editor/controller.js

```diff
@@ -325,3 +326,29 @@ export const redirectSiteEditor = async ( context, next ) => {
 	const siteEditorUrl = addQueryArgs( queryArgs, `${ siteAdminUrl }site-editor.php` );
 	return location.assign( siteEditorUrl );
 };
+/**
+ * Redirect the logged user to the permalink of the post, page, custom post type if the post is published.
+ *
+ * @param {Object} context Shared context in the route.
+ * @param {Function} next  Next registered callback for the route.
+ * @returns {*}            Whatever the next callback returns.
+ */
+export function redirectToPermalinkIfLoggedOut( context, next ) {
+	if ( isUserLoggedIn( context.store.getState() ) ) {
+		return next();
+	}
+	const siteFragment = context.params.site || getSiteFragment( context.path );
+	const CONFIGURABLE_TYPES = [ 'jetpack-portfolio', 'jetpack-testimonial' ];
+	const explodedPath = context.path.split( '/' );
+	if (
+		context.path &&
+		explodedPath[ 1 ] === 'edit' &&
+		! CONFIGURABLE_TYPES.includes( explodedPath[ 2 ] )
```

Comment:
This bit is a bit fragile and dependable on the exact format of the paths. Can we add a comment to clarify with some examples the expected paths this works with?

Reply:
I added a comment about the possible routes